### PR TITLE
feature (route-data): using getChildren where needed

### PR DESCRIPTION
**Pull request by Neill83**
@plentymarkets/team-terra

#### Must be done by Terra

Testing

-   [ ] Person 1 (mandatory)

    Browser-Support (relevant for changes in scss / appearance of components)

    -   [ ] Chrome
    -   [ ] Firefox
    -   [ ] Safari

    Responsiveness

    -   [ ] Desktop
    -   [ ] Tablet
    -   [ ] Mobile

-   [ ] Person 2 (mandatory)

    Browser-Support

    -   [ ] Chrome
    -   [ ] Firefox
    -   [ ] Safari

    Responsiveness

    -   [ ] Desktop
    -   [ ] Tablet
    -   [ ] Mobile

Information transfer

-   [ ] Inform Ceres about changes in `TerraFormComponent` (optional: relevant for changes in this component)

Terra Basic Plugin

-   [ ] Adapt changes (optional: relevant for version updates, global design changes, l10n)

---

#### Must be done by every developer

Please inform a member of Terra (@plentymarkets/team-terra) to get the upper part of the checklist done (with urgency or deadline).

Documentation

-   [ ] Changelog (optional: relevant for every change which influences the API)
-   [ ] Example (updated or created)
-   [ ] JSDoc (optional: relevant for every change which influences the API)
-   [ ] [Google spreadsheet](https://docs.google.com/spreadsheets/d/1OINnux8TEoitV-qdAxqUQaf9oGI_fqFVwfRWenRFfhI/edit#gid=0) (relevant when deprecating public APIs or features)

Testing

-   [ ] Unit Test (updated or created)

### CODE REVIEW

**Review by TimWieder — APPROVED** [review]

File: src/lib/route/utils/routes.ts [hunk 1]
```diff
@@ -2,6 +2,7 @@ import { Route, Routes } from '@angular/router';
 import { RouteData } from '../data/route-data-types';
 import { RouteDataInterface } from '../data/route-data.interface';
 import { normalizeRoutePath } from './route-path';
+import { getChildren } from './route-children-function';
```
Comment (TimWieder) [inline src/lib/route/utils/routes.ts:5]:
```suggestion
import { getChildren } from './route-children';
```

**Review by TimWieder — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,7 @@
 -   **route-data** introducing new `RouteDataModule` and `RouteDataRegistry` that provide functionality to centrally manage additional data for the app's routes.
 -   **breadcrumbs** are now able to retrieve a route's data from the `RouteDataRegistry` when it is not provided with the route itself.
 -   **route** implemented `getChildren()` function which returns the children of a given route whether they are eagerly or lazy loaded.
+-   **get-children** replaced all `route.children` with `getChildren()` method for lazy loading.
```
Comment (pweyrich) [inline CHANGELOG.md:8]: bitte zu den breadcrumbs dazu

Comment (Neill83) [reply CHANGELOG.md:8]: ist nicht nur breadcrumbs

Comment (pweyrich) [reply CHANGELOG.md:8]: doch. der rest ist irrelevant, da noch nicht released.

Comment (pweyrich) [reply CHANGELOG.md:8]: Ich finde, wir sollten auch zeitnah mal drüber sprechen wie wir den Changelog vereinheitlichen. Aktuell ist das mMn etwas chaotisch.

**Review by pweyrich — APPROVED** [review]

**Review by TimWieder — APPROVED** [review]

**Pull request merged**